rascal2/dialogs/project_dialog.py

```diff
@@ -0,0 +1,165 @@
+
+        self.project_name = QtWidgets.QLineEdit(self)
+        self.project_name.setPlaceholderText("Enter project name")
+
+        # Create and style the buttons
+        button_style = """background-color: {};
+                          color: #F2F1E8;
+                          padding-top: 0.3em;
+                          padding-left: 1em;
+                          padding-right: 1em;
+                          padding-bottom: 0.3em;
+                          font-weight: bold;
+                          border-radius: 0.5em"""
+
+        self.browse_button = QtWidgets.QPushButton(" Browse", self)
+        self.browse_button.setIcon(QtGui.QIcon(path_for("open-project-light.png")))
+        self.browse_button.clicked.connect(self.open_folder_selector)
+        self.browse_button.setStyleSheet(button_style.format("#403F3F"))
+
+        self.create_button = QtWidgets.QPushButton(" Create", self)
+        self.create_button.setIcon(QtGui.QIcon(path_for("plus.png")))
+        self.create_button.clicked.connect(self.verify_inputs)
+        self.create_button.setStyleSheet(button_style.format("#0D69BB"))
+
+        self.cancel_button = QtWidgets.QPushButton(" Cancel", self)
+        self.cancel_button.setIcon(QtGui.QIcon(path_for("cancel.png")))
+        self.cancel_button.clicked.connect(self.cancel_project_creation)
+        self.cancel_button.setStyleSheet(button_style.format("#E34234"))
+
+    def cancel_project_creation(self):
+        """
+        Cancel project creation.
+        """
+        if self.parent().new_project_action_called:
+            self.close()
+        else:
+            self.parent().toggleView()
+        self.reset_variables()
+
+    def reset_variables(self):
+        """
+        Resets the variables.
+        """
+        self.folder_selected = False
+        self.project_folder.setText("")
+        self.project_name.setText("")
+
+    def open_folder_selector(self):
+        """
+        Opens the folder selector.
+        """
+        folder_path = QtWidgets.QFileDialog.getExistingDirectory(self, "Select Folder")
+        if folder_path:
+            files_in_folder = list(filter(lambda x: x[0] != ".", os.listdir(folder_path)))
+            if files_in_folder:
+                self.show_error_message(
+                    text="Select a different folder",
+                    info="This folder contains files. Select an empty folder.",
+                    title="Warning",
+                    level="critical",
+                )
+            else:
+                self.project_folder.setText(folder_path)
+                self.folder_selected = True
+
+    def show_error_message(self, text=None, info=None, title=None, level="info"):
+        """
+        Displays the error message.
+        """
+        error_msg = QtWidgets.QMessageBox(self)
+        if level == "critical":
+            error_msg.setIcon(QtWidgets.QMessageBox.Icon.Critical)
+        elif level == "info":
```

Comment:
i'd recommend using the [standard `logging` levels, `logging.CRITICAL` and `logging.INFO`](https://docs.python.org/3/library/logging.html#levels) so that this can hook in nicer with the `logging` library when we add logging
```suggestion
        if level == logging.CRITICAL:
            error_msg.setIcon(QtWidgets.QMessageBox.Icon.Critical)
        elif level == logging.INFO:
```